controller/controller.go

```diff
@@ -445,6 +446,25 @@ func StartInformers(stopCh <-chan struct{}, informers ...Informer) error {
 	return nil
 }
 
+// RunInformers kicks off all of the passed informers in the given error group
+// and then waits for all of them to synchronize.
+func RunInformers(stopCh <-chan struct{}, grp *errgroup.Group, informers ...Informer) error {
```

Comment:
I liked previous approach better.
But the best would probably be to return a call back. Something like

```
func RI(stop, inf...) (cb, error) {
  var g errgroup.Group
  for i := range {
    i:=i
    grp.Go()
  }

  for ... {sync looooooop}
  return func() { wg.Wait()}, nil
```

Now the caller can do
```
ctx, cancel, inf := ....
cb, er := RunInformers(ctx.Done(), inf)
if err { cancel(); handleError }
defer func() {
  cancel()
  cb()
}
cancel()
```